### PR TITLE
Multi cluster release

### PR DESCRIPTION
**Pull request by gokula-krishna-dev**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -119,8 +119,21 @@ commands:
...
           name: Deploying to k8s cluster for service binary-academy
           command: |
-            echo $CA_CRT | base64 --decode > ca.crt
-            kubectl --server=${KUBE_SERVER} --certificate-authority=ca.crt --token=$SERVICEACCOUNT_TOKEN set image deployment/academy-binary-com academy-binary-com=${DOCKHUB_ORGANISATION}/binary-static-academy:${CIRCLE_SHA1}
+            for SERVER_ID in {1..5}
+            do
+              KUBE_SERVER_REF="KUBE_SERVER_$SERVER_ID"
+              SERVICEACCOUNT_TOKEN_REF="SERVICEACCOUNT_TOKEN_$SERVER_ID"
+              CA_CRT_REF="CA_CRT_$SERVER_ID"
+              if [ ! -z "${!KUBE_SERVER_REF}" ]
```
Comment (afshinpaydar-binary) [inline .circleci/config.yml]: why using negation in the IF statement? I think it must be like this:
`if [ -z "${!KUBE_SERVER_REF}" ]`
also, we don't need indirect expansion too:

```
            for SERVER_ID in {1..5}
            do
              KUBE_SERVER_REF="KUBE_SERVER_$SERVER_ID"
              SERVICEACCOUNT_TOKEN_REF="SERVICEACCOUNT_TOKEN_$SERVER_ID"
              CA_CRT_REF="CA_CRT_$SERVER_ID"
              if [  -z "${KUBE_SERVER_REF}" ]
              then
                echo "Deploying to cluster $SERVER_ID"
                CA_CRT="${CA_CRT_REF}"
                KUBE_SERVER="${KUBE_SERVER_REF}"
                SERVICEACCOUNT_TOKEN="${SERVICEACCOUNT_TOKEN_REF}"
                echo $CA_CRT | base64 --decode > ca.crt
                kubectl --server=${KUBE_SERVER} --certificate-authority=ca.crt --token=$SERVICEACCOUNT_TOKEN set image deployment/bitcoin-wallet-binary-com bitcoin-wallet-binary-com=${DOCKHUB_ORGANISATION}/binary-bitcoin-wallet:${CIRCLE_SHA1}
              fi
            done
```

Comment (gokula-krishna-dev) [reply .circleci/config.yml]: `-z` checks for if the string is empty. But what I want is the opposite, I want to check if the variable for `KUBE_SERVER_[i]` is not empty and get the contents of it to proceed with the push to cluster.

Comment (afshinpaydar-binary) [reply .circleci/config.yml]: Yes, you're right about IF statement but what about indirect expansion?

File: .circleci/config.yml [hunk 2]
```diff
@@ -119,8 +119,21 @@ commands:
       - run: 
           name: Deploying to k8s cluster for service binary-academy
           command: |
-            echo $CA_CRT | base64 --decode > ca.crt
-            kubectl --server=${KUBE_SERVER} --certificate-authority=ca.crt --token=$SERVICEACCOUNT_TOKEN set image deployment/academy-binary-com academy-binary-com=${DOCKHUB_ORGANISATION}/binary-static-academy:${CIRCLE_SHA1}
+            for SERVER_ID in {1..5}
```
Comment (mostafa-fs) [inline .circleci/config.yml]: Why it's 1 to 5?

File: .circleci/config.yml (hunk 1 above)
Comment (gokula-krishna-dev) [reply .circleci/config.yml]: On first iteration `SERVER_ID=1`
`"${KUBE_SERVER_REF}"` Will expand to `"KUBE_SERVER_$SERVER_ID"` will expand to `"KUBE_SERVER_1` and then we check if the value of the variable $KUBE_SERVER_1 is not empty.

File: .circleci/config.yml (hunk 2 above)
Comment (gokula-krishna-dev) [reply .circleci/config.yml]: This under the assumption that we will have up to 5 clusters. We can add or remove config easily and we don't have to make any change in code.

**Pull request merged**